lib/topics/state_and_behaviour.rb

```diff
@@ -2,4 +2,59 @@
 
 # exercise state and behaviour
 module StateAndBehaviour
+  # class Car
+  class Car
+    attr_accessor :year, :color, :model, :current_speed
+
+    def self.default_car
+      Car.new
+    end
+
+    def initialize(params)
+      unless params.is_a?(Hash)
+        raise ArgumentError.new("Expects hash of attributes")
```

Comment:
👍 nice and robust. But, in general, it may transform Ruby with its duck-typing principles into some static-typed language.